README.md

```diff
@@ -190,20 +192,115 @@ Shortcut | Description
+`(Monitor) Inset` | A user-configurable screen margin respected by all gTile features (window placement, autogrow, autotiling, …). It causes windows to keep a fixed distance (in pixel) to the monitor edge(s).
+`(Window) Margin` | A user-configurable window margin that causes windows to have an invisible border to them. The margin does not(!) apply towards the screen edges. Windows with a margin are still able to align with the screen edge, i.e., unless an inset is configured.
+
+## Code Structure
 
-For debugging, enable debug in preferences, and in terminal run 
-```shell
-journalctl --follow /usr/bin/gnome-shell | grep gTile
 ```
+dist                - Contents will be distributed 1:1 whenever the extension is build
+├── images          - Assets that are referenced in the CSS stylesheet
+├── schemas         - Contains the GSettings definitions. Required for reading/writing settings
+├── metadata.json   - The extension metadata.json as recognized by Gnome
+└── stylesheet.css  - The stylesheets file as recognized by Gnome
+
+src                 - The TypeScript root directory
+├── core            - Contains the main orchestration classes that are instantiated as singleton
+├── types           - Contains _only_ TypeScript types that to NOT(!) emit code
+├── ui              - Contains reusable UI building blocks utilized by the core classes
+│                     They do not contain any logic other than strictly related to rendering
+├── util            - Reusable helper classes or functions
+├── extension.ts    - The entry point for the extension invoked by Gnome shell
+└── prefs.ts        - The entry point for the preferences dialog
+```
+
+Note that `src/types/` must not contain any files that emit actual JS runtime code. Any files emited by `tsc` during transpilation are deleted by the `postbuild` script in `package.json`.
+
+## Design Principles
+The code base follows the [SOLID](https://en.wikipedia.org/wiki/SOLID) paradigm __up to an extend__. Although it doesn't strictly follow the paradigm it is definitely architectured with these principles in mind. Try to stick with these principles when changing the architecture, e.g., to make [adaptation for different desktop environments](https://github.com/gTile/gTile/issues/103) more easy.
+
+The extension makes use of the GJS mechanisms were possible. In particular, the UI components make extensive use of GObject [Properties and Signals](https://gjs.guide/guides/gobject/basics.html) for synchronization purposes. UI components are modeled as general purpose, composable components. In particular, they do not contain any logic other than strictly related to rendering. Business logic is supposed to reside in an orchestrator.
+
+## Code Style Guide
```

Comment:
Can we reference a widely-used style guide/linter (like [Google's](https://google.github.io/styleguide/tsguide.html)) and enforce that the linter succeeds before letting PRs merge?

Having a few extra bullets with repo-specific preferences seems fine, but a comprehensive style guide is helpful for covering a lot more language cases.

Reply:
I stand by my reasoning for not introducing a linter to the rewrite. But you'd be the person who has more experience with pull requests (and the quality of those) in this repository. So if you come to the conclusion that the contributors aren't capable of sticking with the code conventions I  certainly won't stand in the way of having a linter introduced (and style-guides referenced accordingly).